src/integrationTests/async.spec.ts

```diff
@@ -175,3 +195,165 @@ describe('when window.__uid2.init is called on SdkLoaded from a callback', () =>
     expect(sdkWindow.__uid2.getAdvertisingToken()).toBe(identity.advertising_token);
   });
 });
+
+describe('when getFreshAdvertisingToken is called before init complete', () => {
+  describe('when initialising with a valid identity', () => {
+    const identity = makeIdentity();
+    test('it should resolve promise after invoking the callback', () => {
+      const p = uid2.getFreshAdvertisingToken().then((token: any) => {
+        expect(callback).toHaveBeenCalled();
+        return token;
+      });
+      uid2.init({ callback: callback, identity: identity });
+      jest.runAllTimers();
+      return expect(p).resolves.toBe(identity.advertising_token);
+    });
+  });
+
+  describe('when initialising with an invalid identity', () => {
+    test('it should reject promise after invoking the callback', () => {
```

Comment:
Again here - I'm not sure why the init callback matters if you're testing getFreshAdvertisingToken.